projects/plugins/wpcomsh/feature-plugins/hooks.php

```diff
@@ -0,0 +1,350 @@
+		$disallowed_mimes          = array_merge( $disallowed_mimes, explode( ' ', $upgraded_upload_filetypes ) );
+	}
+
+	if ( ! wpcom_site_has_feature( WPCOM_Features::VIDEOPRESS ) ) {
+		// Copied from WPCOM (see `WPCOM_UPLOAD_FILETYPES_FOR_VIDEOS` in `.config/wpcom-options.php`).
+		// The `ttml` extension is set by `wp-content/mu-plugins/videopress/subtitles.php`.
+		$video_upload_filetypes = 'ogv mp4 m4v mov wmv avi mpg 3gp 3g2 ttml';
+		$disallowed_mimes       = array_merge( $disallowed_mimes, explode( ' ', $video_upload_filetypes ) );
+	}
+
+	foreach ( $disallowed_mimes as $disallowed_mime ) {
+		foreach ( $mimes as $ext_pattern => $mime ) {
+			if ( strpos( $ext_pattern, $disallowed_mime ) !== false ) {
+				unset( $mimes[ $ext_pattern ] );
+			}
+		}
+	}
+
+	return $mimes;
+}
+add_filter( 'upload_mimes', 'wpcomsh_maybe_restrict_mimetypes', PHP_INT_MAX );
+
+/**
+ * Redirect plugins.php and plugin-install.php to their Calypso counterparts if this site doesn't have the
+ * MANAGE_PLUGINS feature.
+ */
+function wpcomsh_maybe_redirect_to_calypso_plugin_pages() {
+	if ( wpcom_site_has_feature( WPCOM_Features::MANAGE_PLUGINS ) ) {
+		return;
+	}
+
+	if ( ! class_exists( 'Automattic\Jetpack\Status' ) ) {
+		return;
+	}
+
+	$request_uri = wp_unslash( $_SERVER['REQUEST_URI'] ); // phpcs:ignore
+
+	$site = ( new Automattic\Jetpack\Status() )->get_site_suffix();
+
+	// Redirect to calypso when user is trying to install plugin.
+	if ( 0 === strpos( $request_uri, '/wp-admin/plugin-install.php' ) ) {
+		wp_safe_redirect( 'https://wordpress.com/plugins/' . $site );
+		exit;
+	}
+
+	if ( 0 === strpos( $request_uri, '/wp-admin/plugins.php' ) ) {
+		wp_safe_redirect( 'https://wordpress.com/plugins/manage/' . $site );
+		exit;
+	}
+}
+add_action( 'plugins_loaded', 'wpcomsh_maybe_redirect_to_calypso_plugin_pages' );
+
+/**
+ * This function manages the feature that allows the user to hide the "WP.com Footer Credit".
+ * The footer credit feature lives in a separate platform-agnostic repository, so we rely on filters to manage it.
+ * Pressable Footer Credit repository: https://github.com/Automattic/at-pressable-footer-credit
+ *
+ * @return bool
+ */
+function wpcomsh_gate_footer_credit_feature() {
+	return wpcom_site_has_feature( WPCOM_Features::NO_WPCOM_BRANDING );
+}
+add_filter( 'wpcom_better_footer_credit_can_customize', 'wpcomsh_gate_footer_credit_feature' );
+
+/**
+ * Remove the Jetpack > Dashboard menu if the site doesn't have the required feature.
+ */
+function wpcomsh_maybe_remove_jetpack_dashboard_menu_item() {
+	if ( wpcom_site_has_feature( WPCOM_Features::JETPACK_DASHBOARD ) ) {
+		return;
+	}
+
+	remove_submenu_page( 'jetpack', 'jetpack#/dashboard' );
+}
+add_action( 'admin_menu', 'wpcomsh_maybe_remove_jetpack_dashboard_menu_item', 1000 ); // Jetpack uses 998.
+
+/**
+ * Remove Jetpack > Manage menu item as part of the wpcom navigation redesign.
+ * For more context, see 5824-gh-Automattic/dotcom-forge.
```

Comment:
No need to shorthand this
```suggestion
 * For more context, see https://github.com/Automattic/dotcom-forge/issues/5824.
```